### PR TITLE
Event updating

### PR DESCRIPTION
**Pull request by JUtupe**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -50,7 +50,9 @@
       "react",
       "@typescript-eslint"
     ],
-    "rules": {},
+    "rules": {
+      "@typescript-eslint/no-empty-function": "off"
```
Comment (DiD3n) [inline package.json]: 👍

File: src/common/models/Event.ts [hunk 2]
```diff
@@ -6,7 +6,7 @@ export class Event {
     constructor(
         public id: string,
         public title: string,
-        public subtitle: string,
+        public subtitle: string | undefined,
```
Comment (DiD3n) [inline src/common/models/Event.ts:9]: only subtitle is optional?

File: src/api/graphql/events/EventCreateMutation.ts [hunk 3]
```diff
@@ -43,10 +64,19 @@ export interface EventRequestInput {
...
+export interface ReplaceEventMutationVars {
+    id: string,
+    request: EventRequestInput
+}
+
+export interface CreateEventMutationData {
     createEvent: EventData
+}
+
+export interface ReplaceEventMutationData {
```
Comment (DiD3n) [inline src/api/graphql/events/EventCreateMutation.ts:75]: we really need to generate types for typescript to end this hell

**Review by DiD3n — APPROVED** [review]

File: src/views/studio/EventCreate/EventCreateForm.tsx [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            onSubmitted={() => navigateUp()}/>
+    } else {
+        return <EventBasicInfoForm
+            event={null}
+            onSubmitted={(createdEvent) => {
+                setEvent(createdEvent)
+                setIndex(1)
+            }}/>
+    }
+}
```
Comment (DiD3n) [inline src/views/studio/EventCreate/EventCreateForm.tsx]: ewww

File: package.json (hunk 1 above)
Comment (JUtupe) [reply package.json]: 👍

File: src/api/graphql/events/EventCreateMutation.ts (hunk 3 above)
Comment (JUtupe) [reply src/api/graphql/events/EventCreateMutation.ts:75]: I added a task for it in trello

File: src/common/models/Event.ts (hunk 2 above)
Comment (JUtupe) [reply src/common/models/Event.ts:9]: fixed

File: src/views/studio/EventCreate/EventCreateForm.tsx (hunk 4 above)
Comment (JUtupe) [reply src/views/studio/EventCreate/EventCreateForm.tsx]: replaced with `switch(inxex)` in new version

**Pull request merged**